tests/Unit/API/Site/Controllers/CountryCodeTraitTest.php

```diff
@@ -0,0 +1,248 @@
+					get_country_code_sanitize_callback as public;
+					get_country_code_validate_callback as public;
+					get_supported_country_code_validate_callback as public;
+				}
+			};
+
+			$this->trait->set_iso3166_provider( $this->iso_provider );
+			$this->trait->set_google_helper_object( $this->google_helper );
+
+			$this->country_supported = true;
+
+			$this->google_helper->method( 'is_country_supported' )
+				->willReturnCallback(
+					function () {
+						return $this->country_supported;
+					}
+				);
+		}
+
+		/**
+		 * Runs after each test is executed.
+		 */
+		public function tearDown(): void {
+			self::$rest_validate_request_arg_callback = null;
+		}
+
+		/**
+		 * Test all get_*_callback methods return a callable.
+		 */
+		public function test_all_get_callbacks_return_callable() {
+			$this->assertIsCallable( $this->trait->get_country_code_sanitize_callback() );
+			$this->assertIsCallable( $this->trait->get_country_code_validate_callback() );
+			$this->assertIsCallable( $this->trait->get_supported_country_code_validate_callback() );
+		}
+
+		/**
+		 * Test the get_country_code_validate_callback and get_supported_country_code_validate_callback
+		 * methods will apply arguments to rest_validate_request_arg_callback and return early if invalid.
+		 */
+		public function test_all_get_validate_callbacks_with_rest_validate_request_arg() {
+			$called_count = 0;
+
+			self::$rest_validate_request_arg_callback = function () use ( &$called_count ) {
+				// When it returns a value other than `true`, it means the validation did not pass,
+				// and here we also make it return a called count.
+				$called_count++;
+				return [ $called_count, func_get_args() ];
+			};
+
+			$args = [ 'US', new Request(), 'country_code' ];
+
+			$callback = $this->trait->get_country_code_validate_callback();
+			$results  = $callback( ...$args );
+
+			$this->assertEquals( [ 1, $args ], $results );
+
+			$callback = $this->trait->get_supported_country_code_validate_callback();
+			$results  = $callback( ...$args );
+
+			$this->assertEquals( [ 2, $args ], $results );
+		}
+
+		/**
+		 * Test a sanitized country code with a string.
+		 */
+		public function test_get_country_code_sanitize_callback_with_string() {
+			$callback = $this->trait->get_country_code_sanitize_callback();
+			$result   = $callback( 'us' );
+
+			$this->assertEquals( 'US', $result );
+		}
+
+		/**
+		 * Test sanitized country codes with an array.
+		 */
+		public function test_get_country_code_sanitize_callback_with_array() {
+			$callback = $this->trait->get_country_code_sanitize_callback();
+			$results  = $callback( [ 'us', 'gb', 'jp' ] );
+
+			$this->assertEquals( [ 'US', 'GB', 'JP' ], $results );
+		}
+
+		/**
+		 * Test a valid country code with a string.
+		 */
+		public function test_get_country_code_validate_callback_with_string() {
+			$callback = $this->trait->get_country_code_validate_callback();
+			$result   = $callback( 'US', new Request(), 'country_code' );
```

Comment:
Should this be `country_codes`? I know this doesn't have an impact on the test but for clarity it would be good to ensure the naming is correct throughout.

Reply:
Probably not. `validate_country_codes` also accepts a `$countries` as a string and this test case is testing it with a string code.

https://github.com/woocommerce/google-listings-and-ads/blob/575d90ee207c4e08400fae8a598355ec69198122/src/API/Site/Controllers/CountryCodeTrait.php#L48

Reply:
Thanks for the clarification and apologies, I thought we were using `country_codes` as the parameter for all endpoints but I can see that's not the case 👍